governance/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,164 @@
+
+_Italics are used to indicate things that need to be replaced._
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | Julian Meyer                                       |
+| Created at         | 2025-02-10                                         |
+| Initial Reviewers  |                  |
+| Need Approval From |                            |
+| Status             | Draft |
+
+> [!NOTE]
+> 📢 Remember:
+>
+> - The single approver in the “Need Approval From” must be from the Security team.
+> - Maintain the “Status” property accordingly. An FMA document can have the following statuses:
+>   - **Draft 📝:** Doc is created but not yet ready for review.
+>   - **In Review 🔎:** Security is reviewing, and Engineering is iterating on the design. A checklist of action items will be created during this phase.
+>   - **Implementing Actions 🛫:** Security has signed off on the content of the document, including the resulting action items. Engineering is responsible for implementing the action items, and updating the checklist.
+>   - **Final 👍:** Security will transition the status of the document to Final once all action items are completed.
+
+> [!TIP]
+> Guidelines for writing a good analysis, and what the reviewer will look for:
+>
+> - Show your work: Include steps and tools for each conclusion.
+> - Completeness of risks considered.
+> - Include both implementation and operational failure modes
+> - Provide references to support the reviewer.
+> - The size of the document will likely be proportional to the project's complexity.
+> - The ultimate goal of this document is to identify action items to improve the security of the project. The FMA review process can be accelerated by proactively identifying action items during the writing process.
+
+## Introduction
+
+This document covers implementation of Pectra features in the OP stack. This requires a hard fork due to new transaction types like EIP-7702 transactions. This does not cover generic hardfork related security concerns, which will be addressed in a separate document.
+
+Included EIPs:
+
+-   EIP-7702: Set code transaction
+-   EIP-2537: BLS12-381 precompiles
+-   EIP-2935: Block hashes contract predeploy
+-   EIP-7251: Increase the MAX_EFFECTIVE_BALANCE
+-   EIP-7623: Increase calldata cost
+-   EIP-7002: Execution layer triggerable withdrawals
+-   EIP-6110: Supply validator deposits on chain
+-   EIP-7685: General purpose execution layer requests
+
+## Failure Modes and Recovery Paths
+
+### FM1: Chain halt due to upgrade transaction failing
+
+- **Description:** The chain could halt if a network upgrade transaction can't be included in the upgrade block.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. end-to-end tests created for network upgrade transactions
+  2. same upgrade path as previous predeploys
```

Comment:
If possible, can we link to other predeploys here so we can be specific in what we're referring to?

Reply:
added links!